### PR TITLE
Added method romFormData to BinaryDownloadPreset to easily use with form data/requests

### PR DESCRIPTION
**Pull request by stefliekens**
This small change can be used in cases where you want to post data/files and you will retrieve a binary download response. E.g. by using the the [Gotenberg APIs](https://gotenberg.dev/docs/routes)

|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no

### CODE REVIEW

File: src/Transport/Presets/BinaryDownloadPreset.php [hunk 1]
```diff
@@ -28,4 +30,19 @@ public static function create(
             BinaryFileDecoder::createWithAutodiscoveredPsrFactories()
         );
     }
+
+    /**
+     * @return TransportInterface<MultiPart, BinaryFile>
+     */
+    public static function fromFormData(
```
Comment (veewee) [inline src/Transport/Presets/BinaryDownloadPreset.php]: Maybe this name is not that good given its context and what it operates upon?
What about : `BinaryDownloadPreset::withMultiPartRequest()`?
Any better alternative is welcome :)


I'm also in doubt about keeping `create()` if we could have a more meaningfull `withEmptyRequest`. (we could alias the `create` to that one for now and deprecate its usage?) What do you think?

**Review by veewee — COMMENTED** [review]
Looks nice, thanks for this!

File: src/Transport/Presets/BinaryDownloadPreset.php (hunk 1 above)
Comment (stefliekens) [reply src/Transport/Presets/BinaryDownloadPreset.php]: I like your renaming suggestions. I cannot come with better names. I'll apply them.

**Comment by stefliekens** [comment]
@veewee, could you check my changes regarding factory method names? Tx.

**Pull request merged**